### PR TITLE
[DUOS-431][risk=no] Handle upload DAA error

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
Alongside https://github.com/DataBiosphere/consent/pull/360, this PR addresses https://broadinstitute.atlassian.net/browse/DUOS-431
Neither PR is dependent upon one another.

## Changes
There exists an edge case where the current profile's DAA is undefined, but the profile is complete. That case was sending invalid data to the API (`undefined` as the url for the existing DAA). This PR fixes this on the front end. 

Additionally, changed a button to a div to prevent unwanted page refreshes from happening underneath the file selection dialog box.

### CODE REVIEW

File: src/pages/ResearcherProfile.js [hunk 1]
```diff
@@ -461,7 +461,8 @@ export const ResearcherProfile = hh(class ResearcherProfile extends Component {
 
         } else {
           if (this.state.file !== undefined && this.state.file.name !== '') {
-            DAR.postDAA(this.state.file.name, this.state.file, profile.urlDAA).then(response => {
+            const existentDAAUrl = (profile.urlDAA === undefined) ? '' : profile.urlDAA;
```
Comment (dags) [inline src/pages/ResearcherProfile.js]: checking only for "undefined" is enough, what about "null" ?

Comment (rushtong) [reply src/pages/ResearcherProfile.js]: Good idea ... lodash has a method for that now: https://lodash.com/docs/4.17.15#isNil

Comment (dags) [reply src/pages/ResearcherProfile.js]: nice !

**Review by dags — APPROVED** [review]

File: src/pages/ResearcherProfile.js (hunk 1 above)
Comment (rushtong) [reply src/pages/ResearcherProfile.js]: There's something else going on here that happened to me earlier in development, but then went away for some reason. Clicking the upload button triggers a refresh of the page so the file upload dialog becomes disconnected to the original form element that triggered it. Will add a WIP to this and dig back in.

**Comment by rushtong** [comment]
@dags Ready for re-review. I updated the surrounding box to a div and styled it so it looked like the button tag that it replaces.

**Review by dags — APPROVED** [review]

**Pull request merged**